api/api/v1alpha1/manifest_types.go

```diff
@@ -35,21 +35,37 @@ type CustomState struct {
 	State      string `json:"state,omitempty"`
 }
 
-// ChartInfo defines chart information
-type ChartInfo struct {
-	ChartPath    string `json:"chartPath,omitempty"`
-	RepoName     string `json:"repoName,omitempty"`
-	Url          string `json:"url,omitempty"`
-	ChartName    string `json:"chartName,omitempty"`
-	ReleaseName  string `json:"releaseName,omitempty"`
-	ClientConfig string `json:"clientConfig,omitempty"`
-	Overrides    string `json:"overrides,omitempty"`
+// InstallInfo defines installation information
+type InstallInfo struct {
+	ConfigInfo `json:",inline"`
+	// +kubebuilder:validation:Enum=HelmChart;Executable
```

Comment:
Double checked this again, currently not easily possible since I can only tell if you that its of type localOciBlob without introspecting the metadata.

Reply:
I recommend to leave validation out and assume its a helm chart directory